group-membership/src/Code.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Trigger that runs on edit after being installed via the interface.
+ *
+ * @param {Object} e - onEdit trigger event.
+ */
+function onEditInstallableTrigger(e) {
+  // Get the headers, row range and values from the active sheet.
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var headers = sheet.getDataRange().offset(0, 0, 1).getValues()[0];
+  var range = sheet.getRange(e.range.getRow(), 1, 1, headers.length);
+  var row = range.getValues()[0];
+
+  // Convert the row Array into an entries Object using the headers for the
+  // field names.
+  var entries = headers.reduce(function(result, columnName, i) {
+    result[columnName] = row[i];
+    return result;
+  }, {});
+
+  // Update the entries Object with the status returned by addToGroup().
+  try {
+    entries['Status'] = addToGroup(
+        entries['Email'],
+        entries['Google Group'],
+        entries['Allowed'],
+        entries['Email template doc URL'],
+        entries['Email subject']
+    );
+  } catch (e) {
+    // If there's an error, report that as the status.
+    entries['Status'] = e;
+  }
+
+  // Convert the updated entries Object into a row Array.
+  var rowToWrite = headers.map(function(columnName) {
+    return entries[columnName];
+  });
+
+  // setValues() receives a 2D array, so we create an array with the row
+  // contents.
+  range.setValues([rowToWrite]);
+}
+
+
+/**
+ * Trigger that runs on edit after being installed via the interface.
+ *
+ * @param {string} userEmail - email of user to add to the group.
+ * @param {string} groupEmail - address of Google Group to add user to.
+ * @param {string} allowed - 'yes' flag to add user to group.
+ * @param {string} emailTemplateDocUrl - Google Doc URL that serves as template
+ *        of the welcome email sent to a user added to the group.
+ * @param {string} emailSubject - subject of welcome email sent to user added to group.
+ * @return {string} - status if email was sent to a user added in the sheet.
```

Comment:
if you're returning a binary status it should be a boolean, strings are brittle

Reply:
Thank you for the suggestion. The function can return up to 3 different status:
1) 'Sent'+ date
2) 'Already in group
3) 'Not sent'
Since I am not using the return status for any further processing, it's only for user feedback, so I think it's ok if the string changes.

Reply:
The status returned are now constant strings instead of arbitrary strings. Couldn't do boolean since it only returns 2 states, and this solution offers 3.